site/js/modules/webui/event.js

```diff
@@ -0,0 +1,111 @@
+// A package to listen for custom events.
+// An event listener listens for custom events with the same name as the listener to be dispatched.
+// All custom events may provide additional details which will be stored in event.detail.
+// To listen for a specific event within an event type, provide additional event details.
+// Every key-value pair in the provided details must be matched in the event.detail information.
+// To check for object equality, each value is compared via JSON.stringify().
+// If the value cannot be stringified, basic equality (===) will be used for the comparison.
+
+// Keyed on the listener ID with a reference to the event name and callback function.
+const eventListeners = new Map();
+const eventElement = document.createElement(`div`);
+
+const EVENT_TYPE_HANDLER_COMPLETED = 'Event Type: Handler Completed';
```

Comment:
This name seems strange for a couple reasons:
 - Is the "Event Type" prefix required? Other JS events don't have a prefix.
 - I believe this is the handle for when routing is complete, not just some random handler.